### PR TITLE
feat: load resolver from user configuration file

### PR DESCRIPTION
**Pull request by tomrav**
This PR causes the extension to on load look for a `stylable.config.js` file in the root of the project.
If it finds one, it'll attempt to load the resolver (if exists), and use it to create it's instances of Stylable and its LSP.

There's a technical debt here where the file name is hard-coded for the time being, where all of our other integrations receive can receive a file path location. This might be something we can extract from the user extension configuration in VSCode, but will need further exploring.

### CODE REVIEW

File: src/lib/server.ts [hunk 1]
```diff
@@ -13,24 +13,53 @@ import { initializeResult } from './capabilities';
...
+    let resolveModule;
+
+    console.log('rootUri', rootUri);
+
+    console.log('configPath', configPath);
+    console.log('rootFsPath', rootFsPath);
+
+    try {
+        if (configPath) {
+            const { defaultConfig } = (await import(configPath)) as {
```
Comment (barak007) [inline src/lib/server.ts:39]: not sure about this asyncness here. does on onInitialize wait for it? 

it delay the creation of `vscodeStylableLSP` we need to double check that we don't have a race here.

do we want to keep these console.logs?

File: src/lib/server.ts [hunk 2]
```diff
@@ -13,24 +13,53 @@ import { initializeResult } from './capabilities';
...
+            };
+
+            resolveModule =
+                defaultConfig && typeof defaultConfig === 'function' ? defaultConfig(fs).resolveModule : undefined;
+
+            console.log('defaultConfig', defaultConfig(fs));
+            console.log('resolveModule', resolveModule);
+        }
+    } catch (e: unknown) {
+        console.log(new Error(`Failed to load Stylable config from ${configPath || 'UNKNOWN PATH'}:\n${e as string}`));
```
Comment (barak007) [inline src/lib/server.ts:50]: this should be console.warn semantically and also tell the user that you continue with the default config instead.
and extract to function.

File: src/lib/server.ts (hunk 1 above)
Comment (tomrav) [reply src/lib/server.ts:39]: From what I could see in its types, I think it accepts a function that can return a `Thenable`. While playing around with it I didn't notice any odd behavior. 

Left extra logs here, cleaning up.

Comment (barak007) [reply src/lib/server.ts:39]: `Thenable` is good start!

File: src/lib/server.ts (hunk 2 above)
Comment (tomrav) [reply src/lib/server.ts]: done

**Pull request merged**